### PR TITLE
xds: env var protection for xds fallback

### PR DESCRIPTION
**Pull request by easwars**
Fixes https://github.com/grpc/grpc-go/issues/7479

RELEASE NOTES: none

### CODE REVIEW

File: internal/xds/bootstrap/bootstrap_test.go [hunk 1]
```diff
@@ -1194,3 +1198,78 @@ func (s) TestNode_ToProto(t *testing.T) {
 		})
 	}
 }
+
+// Tests the case where the xDS fallback env var is set to false, and verifies
+// that only the first server from the list of server configurations is used.
+func (s) TestGetConfiguration_FallbackDisabled(t *testing.T) {
+	// TODO(easwars): Default value of "GRPC_EXPERIMENTAL_XDS_FALLBACK"
+	// env var is currently false. When the default is changed to true,
+	// explicitly set it to false here.
```
Comment (arvindbr8) [inline internal/xds/bootstrap/bootstrap_test.go:1207]: optional: explicitly state what the test expects?

**Review by arvindbr8 — APPROVED** [review]
LGTM

File: internal/xds/bootstrap/bootstrap_test.go (hunk 1 above)
Comment (easwars) [reply internal/xds/bootstrap/bootstrap_test.go:1207]: This whole test is expected to be run with the env var set to false. This is stated in the name of the test and the top-level comment.

Comment (arvindbr8) [reply internal/xds/bootstrap/bootstrap_test.go:1207]: i meant

```
			origFallbackEnv := envconfig.XDSFallbackSupport
			envconfig.XDSFallbackSupport = false
			defer func() { envconfig.XDSFallbackSupport = origFallbackEnv }()
```

Comment (easwars) [reply internal/xds/bootstrap/bootstrap_test.go:1207]: Ah sorry. Should have done that. But now, I will just wait for the default value to change before doing that :)

**Pull request merged**